### PR TITLE
Fix deadcode issues using golangci-lint

### PR DESCRIPTION
**Pull request by agnivade**

### CODE REVIEW

**Comment by DHaussermann** [comment]
@hanzei  or @agnivade can you please advise if there is any targeted regression testing suitable for this change?

**Comment by agnivade** [comment]
Will let Ben/Maria take a call.

**Comment by hanzei** [comment]
Because this is removing code that is not used, i'm very confident that there will not introduce regressions. A general smoke test is enough.

File: server/plugin.go [hunk 1]
```diff
@@ -24,20 +24,16 @@ import (
...
 	GITHUB_TOKEN_KEY        = "_githubtoken"
-	GITHUB_STATE_KEY        = "_githubstate"
 	GITHUB_USERNAME_KEY     = "_githubusername"
 	GITHUB_PRIVATE_REPO_KEY = "_githubprivate"
 	WS_EVENT_CONNECT        = "connect"
 	WS_EVENT_DISCONNECT     = "disconnect"
 	WS_EVENT_REFRESH        = "refresh"
 	SETTING_BUTTONS_TEAM    = "team"
-	SETTING_BUTTONS_CHANNEL = "channel"
-	SETTING_BUTTONS_OFF     = "off"
```
Comment (marianunez) [inline server/plugin.go]: These are also defined in the webapp here : https://github.com/mattermost/mattermost-plugin-github/blob/f67b80b3daca4b99b036b59759148e241ac44c2a/webapp/src/constants/index.js#L6-L7

We should remove them there to be consistent.

**Pull request merged**